### PR TITLE
Add folder structure and naming conventions

### LINKED ISSUE #19: [Feature Request]: Folder structure and naming convention
### Feature details

Add folder structure and naming convention

### Suggested Solution

Refer to [this repo](https://github.com/DevSquadIN/nextjs14-folder-structure)

### Affected areas

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Rick799**
Added folder structure and naming conventions

Fixes #19 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots or example output

### CODE REVIEW

**Review by joypoddar — CHANGES_REQUESTED** [review]

**Comment by joypoddar** [comment]
Good job @Rick799 . There are minor changes. Take a look and resolve them

File: README.md [hunk 1]
```diff
@@ -44,6 +44,103 @@ npx create-next-app -e https://github.com/CreoWis/next-js-launchpad
...
+    ├── modules
+    ├── queries
+    ├── services
+    ├── store
+    ├── utils
+├── styles
+├── types
+```
+
+\***\*tests\*\***: This folder contains all 'tests'.
```
Comment (joypoddar) [inline README.md:72]: image
Doesn't appear right. please correct it.

File: README.md [hunk 2]
```diff
@@ -44,6 +44,103 @@ npx create-next-app -e https://github.com/CreoWis/next-js-launchpad
...
+  } as const satisfies OrderStatus;
+```
+
+- component names -> `PascalCase` E.g. _export const Header = () => {}_
+- types should have `.d.ts` extension. E.g. _root.d.ts_
+- use arrow functions wherever possible.
+- ui folder contains granular level components like _button_, _dialog_, _input_ etc which can be used to build other components.
+- use given folders for various kinds of files. E.g. _types should reside in types folder_
+
+For granular level convention, read the README in specific folder.
```
Comment (joypoddar) [inline README.md:140]: Delete this line

File: README.md [hunk 3]
```diff
@@ -44,6 +44,103 @@ npx create-next-app -e https://github.com/CreoWis/next-js-launchpad
...
+
+NOTE: Check specific folder README for overiding rules (if any). If there are cases where industry standards over-ride the rules mentioned above, please feel free to use them. For example convention for naming hooks.
```
Comment (joypoddar) [inline README.md:142]: We no longer have README in specific folders. Re-word this sentence to accommodate the current scenario.

File: src/config/folder.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
+export const FOLDER_CONFIG = [
```
Comment (joypoddar) [inline src/config/folder.ts:1]: Check if this file is being used anywhere. If not please remove it.

File: README.md [hunk 5]
```diff
@@ -44,6 +44,103 @@ npx create-next-app -e https://github.com/CreoWis/next-js-launchpad
...
+## Folder Structure
+
+```bash
+.
+├── __tests__
+├── lib
+├── public
+└── src
+    ├── app
+    ├── components
```
Comment (joypoddar) [inline README.md:56]: Please put the subfolder under components

**Review by joypoddar — CHANGES_REQUESTED** [review]

File: src/data/README.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# src/context
```
[the remark this answers is not in the document]
Comment (joypoddar) [reply src/data/README.md]: closed on str change

File: src/db/README.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# src/context
```
[the remark this answers is not in the document]
Comment (joypoddar) [reply src/db/README.md]: closed on str change

**Review by joypoddar — APPROVED** [review]

**Review by joypoddar — APPROVED** [review]

**Review by atapas — APPROVED** [review]

**Pull request merged**